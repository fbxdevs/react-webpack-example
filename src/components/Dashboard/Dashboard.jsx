import React from 'react';
import {Component} from 'react';
import {Container, Row, Col} from 'reactstrap';

/**
 * Dashboard class extending react.Component
 */
class Dashboard extends Component {
  /**
   * Class constructor
   * required
   * @param {object} props passed to Class
   */
  constructor(props) {
  	super(props);
  	this.state = {};
  }

   /**
    * componentWillMount is invoked before mounting
    * not required
    */
   componentWillMount() {}

   /**
    * render examins this.props and this.state to return a react element,
    * string and numbers, portals, null and booleans
    * required
    * @return {ReactElement} markup
    */
   render() {
     return (
       <Container>
				 <Row>
					 <Col md="12">
						 <b>Content!</b>
					 </Col>
				 </Row>
			 </Container>
     );
   }

   /**
    * componentDidMount is invoked after mounting
    * not required
    */
   componentDidMount() {}

   /**
    * componentWillReceiveProps is invoked before
    * a mounted component receives new props
    * not required
    * @param {object} nextProps the next properties to attach to component
    */
   componentWillReceiveProps() {}

   /**
    * shouldComponentUpdate is to let React know if a componentâ€™s
    * output is not affected by the current change in state or props
    * not required
    * @param {object} nextProps the next properties to attach to component
    * @param {object} nextState the next state of component
    * @return {bool} update component or not
    */
   shouldComponentUpdate() {
     return true;
   }

   /**
    * componentWillUpdate is invoked just before rendering
    * when new props or state are being received
    * Use this as an opportunity to perform preparation before an update
    * if shouldComponentUpdate is false componentWillUpdate is ignored
    * not required
    * @param {object} nextProps the next properties to attach to component
    * @param {object} nextState the next state of component
    */
   componentWillUpdate() {}

   /**
    * componentDidUpdate is invoked immediately after updating occurs
    * if shouldComponentUpdate is false componentDidUpdate is ignored
    * not required
    * @param {object} prevProps the previous properties attached to component
    * @param {object} prevState the previous state of component
    */
   componentDidUpdate() {}

   /**
    * componentWillMount is invoked immediately before
    * a component is unmounted and destroyed
    * not required
    */
   componentWillUnmount() {}

   /**
    * error handling
    * not required
    * @param {object} error
    * @param {object} info
    */
   componentDidCatch() {}
}

Dashboard.defaultProps = {

};

Dashboard.displayName = 'Dashboard';

export default Dashboard;
