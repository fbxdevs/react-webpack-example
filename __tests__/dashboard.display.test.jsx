// Testing elements coming from DB displaying on dashboard
import React from 'react';
import Dashboard from '../src/components/Dashboard';
import {shallow} from 'enzyme';

test('Render dashboard with a bootstrap row and column', () => {

  const component = shallow(
		<Dashboard />
  );

  // check if it matches snapshot of rendered item
	expect(component).toMatchSnapshot();
});
