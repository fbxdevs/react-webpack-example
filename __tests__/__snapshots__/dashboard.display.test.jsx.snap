// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render dashboard with a bootstrap row and column 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row
        tag="div"
      >
        <Col
          md="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <b>
            Content!
          </b>
        </Col>
      </Row>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          md="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <b>
            Content!
          </b>
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <b>
            Content!
          </b>,
          "md": "12",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Content!",
          },
          "ref": null,
          "rendered": "Content!",
          "type": "b",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <Col
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <b>
              Content!
            </b>
          </Col>
        </Row>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <b>
              Content!
            </b>
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <b>
              Content!
            </b>,
            "md": "12",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Content!",
            },
            "ref": null,
            "rendered": "Content!",
            "type": "b",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
